plugins/tech-insights-backend/src/service/fact/factRetrievers/catalogFactRetriever.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  FactRetriever,
+  FactRetrieverContext,
+} from '@backstage/plugin-tech-insights-node';
+import { CatalogClient } from '@backstage/catalog-client';
+import { Entity } from '@backstage/catalog-model';
+import isEmpty from 'lodash/isEmpty';
+import camelCase from 'lodash/camelCase';
+import { get } from 'lodash';
+
+type Options = {
+  annotations?: string[];
+};
+
+export const createCatalogFactRetriever = (
+  { annotations = [] }: Options = { annotations: [] },
+): FactRetriever => ({
+  id: 'entityRetriever',
+  version: '0.0.1',
+  schema: {
+    hasOwner: {
+      type: 'boolean',
+      description: 'The spec.owner field is set',
+    },
+    hasGroupOwner: {
+      type: 'boolean',
+      description: 'The spec.owner field is set and refers to a group',
+    },
+    hasDescription: {
+      type: 'boolean',
+      description: 'The entity has a description in metadata',
+    },
+    hasTags: {
+      type: 'boolean',
+      description: 'The entity has tags in metadata',
+    },
+    ...annotations.reduce((acc: object, annotation: string) => {
```

Comment:
It occured to me that having dynamic values here renders versioning largely meaningless so I'm going to remove this in favour of explicit annotation checks. That being the case I'm going to split this into three:
EntityOwnershipFactRetriever, EntityMetadataFactRetriever and TechdocsFactRetriever (naming might be more specific given it's only going to check the annotation).